src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -683,6 +712,14 @@ module.exports = function<T, P, I, TI, PI, C, CX>(
     // Unmount the current children as if the component rendered null
     const nextChildren = null;
     reconcileChildren(current, workInProgress, nextChildren);
+
+    if (workInProgress.tag === ClassComponent) {
+      const instance = workInProgress.stateNode;
+      workInProgress.memoizedProps = instance.props;
+      workInProgress.memoizedState = instance.state;
```

Comment:
I still don't understand why this needs to be done in `beginFailedWork` specifically.

If anything, this should set the memoized values to `null` since this component is now in an invalid state. The children output doesn't correspond to the input.